schema/jsonschema.json

```
@@ -0,0 +1,81 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "dispatches": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "type": {
+            "type": "string",
+            "enum": ["snapshots", "releases"]
+          },
+          "flavors": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "minItems": 1
+          },
+          "state_repos": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "properties": {
+                "repo": {
+                  "type": "string",
+                  "minLength": 1
+                },
+                "tenant": {
+                  "type": ["string", "null"]
+                },
+                "application": {
+                  "type": "string",
+                  "minLength": 1
+                },
+                "env": {
+                  "type": "string",
+                  "minLength": 1
+                },
+                "service_names": {
+                  "type": "array",
+                  "items": {
+                    "type": "string"
+                  },
+                  "minItems": 1
+                },
+                "version": {
+                  "type": "string",
+                  "minLength": 1
+                },
+                "registry": {
+                  "type": "string",
+                  "minLength": 1
+                },
+                "dispatch_event_type": {
+                  "type": "string",
+                  "default": "dispatch-image"
+                }
+              },
+              "required": [
+                "repo",
+                "tenant",
+                "application",
+                "env",
+                "service_names",
+                "version"
+              ],
+              "additionalProperties": true
```

```suggestion
              "additionalProperties": false
```

All properties should be controlled by the json schema, not additional properties allowed